corehq/apps/cloudcare/esaccessors.py

```diff
@@ -72,4 +74,14 @@ def login_as_user_query(
         ).location_ids()
         user_es = user_es.location(list(loc_ids))
 
+    if not couch_user.has_permission(domain, get_permission_name(Permissions.login_as_all_users)):
+        user_es = user_es.filter(
+            filters.nested(
+                'user_data_es',
+                filters.AND(
+                    filters.term('user_data_es.key', 'login_as_user'),
+                    filters.term('user_data_es.value', couch_user.username),
```

Comment:
Confirming: The user doesn't have any ability to change this username, right? This is the, like, permanent HQ one that can't change?

Reply:
The value in this filter is the permanent HQ one, but the value saved to the mobile worker is just custom user data, so that is configurable.